drivers/mmc/host/sdhci.c

```diff
@@ -3627,7 +3633,7 @@ static irqreturn_t sdhci_irq(int irq, void *dev_id)
 		if (intmask & SDHCI_INT_RETUNE)
 			mmc_retune_needed(host->mmc);
 
-		if (intmask & SDHCI_INT_TUNING_ERR) {
+		if (intmask & SDHCI_INT_TUNING_ERROR) {
```

Comment:
I believe our out-of-tree patch https://github.com/ni/linux/commit/13dc7b7f158917dec94d79bdd4abea66f4f7e6ae and the new upstream patch https://github.com/torvalds/linux/commit/b3855668d98cf9c6aec2db999dd27d872f8ba878 essentially accomplish the same thing, i.e. re-tune if a SDHCI_INT_TUNING_ERROR interrupt is received:

```
if (intmask & SDHCI_INT_TUNING_ERROR) {
	u16 ctrl2 = sdhci_readw(host, SDHCI_HOST_CONTROL2);

	ctrl2 &= ~SDHCI_CTRL_TUNED_CLK;
	sdhci_writew(host, ctrl2, SDHCI_HOST_CONTROL2);
}

```

By keeping both patches you will end-up triggering the re-tune twice, once from our out-of-tree patch here and before this via the newly added sdhci_data_irq() code called from a couple of lines above.

We should double check, but it looks likely we can drop our out-of-tree patch in favor of the patch that's already upstream. Our patch has an extra check and error print but I don't think that's consequential.

My recommendation would be to see if we can revert our out-of-tree patch: https://github.com/ni/linux/commit/13dc7b7f158917dec94d79bdd4abea66f4f7e6ae, do the upstream merge (which should be conflict free after revert) and test if the original problem is solved by the upstream patch. The original author of our out-of-tree patch, Kyle Roeschley should be able to help. It would be great if we don't have to carry this patch anymore.

Reply:
Btw, for WI: [AB#2316627](https://dev.azure.com/ni/DevCentral/_workitems/edit/2316627), we setup the cRIO-9053 in our rack with a swissbit card that's required to reproduce some of the bugs in this mmc patch series.

The card seems to no longer plugged into the 9053 but it's probably somewhere around it - one of our technicians can help plug it in.

Reply:
I am able to revert the out-of-tree patch and do the upstream merge and did not encounter any conflicts this time. I am able to rebuild the kernel and did a basic sanity check, but not sure how to test only those specific changes that we pulled from [upstream commit](https://github.com/torvalds/linux/commit/b3855668d98cf9c6aec2db999dd27d872f8ba878). I have pinged Kyle Roeschley to get help for testing the changes. Will update here once the new changes work.